src/tests/ftest/launch.py

```diff
@@ -790,13 +788,37 @@ def run_tests(test_files, tag_filter, args):
         command_list.extend(tag_filter)
 
     # Run each test
+    skip_reason = None
     for test_file in test_files:
-        if isinstance(test_file["yaml"], str):
+        if skip_reason is not None:
+            # An error was detected running clean_logs for a previous test.  As
+            # this is typically an indication of a communication issue with one
+            # of the hosts, do not attempt to run subsequent tests.
```

Comment:
The alternative here is to try to correct the nodes that caused failures in collecting logs.  Indeed, that is probably something that should be done in the log collection, so as to have a higher chance of getting the logs in such a situation, yes?

That would be preferable to failing the test run wouldn't it?

Reply:
I think something like what you're proposing would be best dealt with in another ticket.   This PR is addressing the lack of notification for tests not run - not so much as to why it was not run in the first place.  

Assuming the communication error when running `clush sudo rm -fr <logs>` was not intermittent it will be hard to get anything form the host, unless we're talking about verbose ssh command output.

Reply:
Yes, another ticket perhaps.  But just to be clear, when remaining tests are not run due to previous errors collecting logs, that test run fails, yes?  And IIRC, produces a JUnit result indicating such.

> Assuming the communication error when running clush sudo rm -fr <logs> was not intermittent it will be hard to get anything form the host, unless we're talking about verbose ssh command output.

￼I'm talking about taking proactive action to get the host to a state where it can be communicated with again.  Attempting a reboot, power-cycle failing that, etc.

Reply:
> Yes, another ticket perhaps. But just to be clear, when remaining tests are not run due to previous errors collecting logs, that test run fails, yes? And IIRC, produces a JUnit result indicating such.

It's not a case of the error log collection gone wrong - in that case we just continue on to the next test.  

This PR is concerning the issue where prior to running a test (`avocado run <file>`) we remove any logs that may exist on the remote hosts (that could have been left over from a failed archive command).  If `launch.py` detects an issue communicating with any of theses hosts it does not execute this test or any other tests that remain in the list:
```
[2020-10-16T18:06:20.820Z] Cleaning logs on ['wolf-137', 'wolf-138', 'wolf-139', 'wolf-140', 'wolf-141']
[2020-10-16T18:06:20.820Z] Running on wolf-[137-141]: sudo rm -fr /var/tmp/daos_testing/*.log*
[2020-10-16T18:06:24.149Z]   Errors detected running "sudo rm -fr /var/tmp/daos_testing/*.log*":
[2020-10-16T18:06:24.149Z]     wolf-[137-138,140-141]: rc=0, output: <NONE>
[2020-10-16T18:06:24.149Z]     wolf-139: rc=255, output: ssh: connect to host wolf-139 port 22: No route to host
[2020-10-16T18:06:24.149Z] Error cleaning logs, aborting
[2020-10-16T18:06:24.149Z] ERROR: Failed to clean logs in preparation for test run!
```

The problem is `launch.py` does not clearly indicate which tests it did not execute.  The changes in this PR generate individual JUnit results for those tests so the failure to execute the tests is clearly indicated.


> I￼'m talking about taking proactive action to get the host to a state where it can be communicated with again. Attempting a reboot, power-cycle failing that, etc.

Ah, ok.  Seeing as that is new behavior for `launch.py` I think it should be a new PR. 
 
> Is there a test run in the Jenkins history for this PR that simulates such a failure, to be sure that the code works as intended? I'd be interested in seeing what the results look like if there is.

The details are in https://jira.hpdd.intel.com/browse/DAOS-5874

Reply:
> This PR is concerning the issue where prior to running a test (`avocado run <file>`) we remove any logs that may exist on the remote hosts (that could have been left over from a failed archive command). If `launch.py` detects an issue communicating with any of theses hosts it does not execute this test or any other tests that remain in the list

Ahh.  Yes, I see now.

> The problem is `launch.py` does not clearly indicate which tests it did not execute. The changes in this PR generate individual JUnit results for those tests so the failure to execute the tests is clearly indicated.

Got it.

> Ah, ok. Seeing as that is new functionality for `launch.py` I think it should be a new PR.

Agreed.

> > Is there a test run in the Jenkins history for this PR that simulates such a failure, to be sure that the code works as intended? I'd be interested in seeing what the results look like if there is.
> 
> The details are in https://jira.hpdd.intel.com/browse/DAOS-5874

Yeah, that ticket describes the issue well.  I'm just wondering if you simulated the error condition this PR is trying to address in any of the runs of the PR, to be sure that the code works.  I'd like to see the results of that run, if you did one.  I can certainly imagine what the result would be, but seeing it would confirm the user-experience.

Reply:
https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-3699/5/ includes a small change to have the `launch.py` `clean_logs()` method randomly return False (even when the remote log removal succeeds) to:
-  force the skipping of the active test
-  force the skipping of any remaining tests
-  create a JUnit file for each test indicating it was skipped

Reply:
Sorry I didn't see your response (page didn't refresh) until I posted my response.  I do agree this sort of test is needed to ensure the changes can be committed.  Since build 4 was successful (and assuming build 5 works as intended), I'm hoping that a commit back to the build 4 version of `launch.py` could just be force landed w/o a CI run.

Reply:
Yeah, I will sometimes create debug/test PRs and base them on the PR I am debugging/testing with the debug/test changes I need to avoid polluting the main PR.  Then discard them when done with them.

When doing debug/tests, don't forget to take advantage of `Skip-*`s, particularly for hardware stages to avoid unnecessarily tying up cluster resources.

Reply:
https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-3699/9/artifact/Functional/ includes examples of JUnit files generated for test that have been skipped:

![image](https://user-images.githubusercontent.com/48695441/97740828-30cd5600-1ab8-11eb-85dd-b273e1efdfd8.png)

With a failed test report that looks similar to this:

![image](https://user-images.githubusercontent.com/48695441/97741140-ad603480-1ab8-11eb-9159-2d7a3eb21880.png)